### PR TITLE
Move source files to src/

### PR DESCRIPTION
**Pull request by marvinhagemeister**
This makes generating compilation targets easier as we don't have to maintain a list of files anymore and can just feed a `src/*.js` glob to compilers.

Due to files being moved around the summary diff is pretty hard to follow. Reviewing this PR on a commit-by-commit basis is a lot easier.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,86 +2,86 @@
   "name": "pentf",
   "version": "1.0.95",
   "description": "parallel end-to-end test framework",
-  "main": "index.js",
-  "types": "index.d.ts",
+  "main": "src/index.js",
+  "types": "types/index.d.ts",
```
Comment (jzaefferer) [inline package.json:6]: Is this correct? The clean task doesn't touch this, only `docs` and `dist`, and the `tsconfig.json` change references `"declarationDir": "dist/types/",`

File: tests/selftest_browser_close.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    await page.setContent('Hello world');
+
+    onTeardown(config, () => {
+        // console.log(page)
+        assert(page._closed, 'Page wasn\'t closed during teardown');
+    });
+}
+
+module.exports = {
+    description: 'Close browser automatically on teardown',
```
Comment (jzaefferer) [inline tests/selftest_browser_close.js:16]: This test sounds reasonable, but is rather unexpected in this PR. Was the file added by accident?

File: package.json (hunk 1 above)
Comment (marvinhagemeister) [reply package.json:6]: Good catch! You're right, it should be `dist/types/index.d.ts` 👍

File: tests/selftest_browser_close.js (hunk 2 above)
Comment (marvinhagemeister) [reply tests/selftest_browser_close.js]: Whoops, yup that was an accident. Looks like I'm switching between too many branches 🙈
